app-frontend/src/app/components/scenes/sceneDetailModal/sceneDetailModal.html

```diff
@@ -9,20 +9,33 @@
           <input type="text" class="form-control"
                  ng-attr-title="{{$ctrl.scene.name}}" ng-model="$ctrl.scene.name" readonly>
         </dl>
+      </div>
+      <div class="scene-metadata-scrollable">
         <dl>
-          <strong>{{$ctrl.datasource ? $ctrl.datasource : 'Loading datasource'}}</strong>
-        </dl>
-        <dl>
           <button class="btn btn-default"
-                  ng-click="$ctrl.toggleMetadataEdit()"
-                  ng-if="$ctrl.scene.owner === $ctrl.authService.getProfile().sub">
-            {{$ctrl.isEditMetadata ? 'Save ' : 'Edit '}}metadata
+                  ng-click="$ctrl.startEditing()"
+                  ng-if="!$ctrl.editingMetadata && $ctrl.isOwner">
+            Edit metadata
+          </button>
+          <button class="btn btn-light"
+                  ng-click="$ctrl.cancelEditing()"
+                  ng-if="$ctrl.editingMetadata">
+            Revert changes
```

Comment:
This should probably be 'Cancel' since that is the wording we use elsewhere in the UI.

Reply:
This would also resolve this problem with the high-contrast theme:
![image](https://user-images.githubusercontent.com/2442245/36104084-07b8b188-0fdf-11e8-964c-788b5696aea4.png)